### ISSUE TITLE
Syncthing source contains minified js

### ISSUE THREAD

**Issue by anoteng** [syncthing/syncthing#1560]
Hello. I'm packaging syncthing for Debian and will try to get it into the main archive.
In Debian I'm not allowed to ship minified js in the binary or source packages without having the full source in the source package. Debian does not consider minified js valid source code.
The ideal solution to this is in my eyes to include the full sources in the syncthing source and add steps to minify in the build process. This would also make it easier for anyone reviewing the syncthing source code to verify that the js libs are indeed what they claim to be and that they do not contain malicious code.
 
My other option would be to repackage the source tarball for the Debian package and add the full js source and minify during the build myself.

Cheers!

**Comment by calmh** [comment]
It's worth than that. All JS and HTML is gzipped and base64 encoded into the `internal/auto` package which is then compiled into the binary.

There's already a `./build.sh assets` (`go run build.go assets`) build step that's used to do the above, currently based on the minified JS etc. I would accept a pull request that used non-minified JS sources and minified them as part of that build step.

(I'm less sure about how to handle the dependencies required for that though, but that's a detail.)

**Comment by anoteng** [comment]
I'll see what I can do. I'll prioritize making it work for Debian in a way that is compatible with the Debian free software guidelines.

**Comment by calmh** [comment]
The easiest thing to try may be to just use the non-minified jquery and angular sources. Since we anyway gzip them into the binary, it may not make any difference whatsoever to the binary size or memory usage.

**Comment by anoteng** [comment]
It might make a difference to the size of the files sent to the browser though?

**Comment by calmh** [comment]
They're sent gzipped in pretty much every case (now, since a recent change), and the user is probably usually local anyway, so I wouldn't worry much about it.

**Comment by anoteng** [comment]
I solved this for Debian by adding the unminified concatenated js-files from the updstream projects to debian/missing-sources/. During package build I use "uglifyjs filname.js > gui/vendor/filname.min.js" and finish off with "./build.sh assets" before compiling.
I'll send a patch when I have time, but for bootstrap I used the package allready available in Debian instead of minifying it myself. The Debian package of bootstrap also remove the inluded fonts.
